# Is there disporpotionate disciplinary action in CMS?

This document strives to explore differences in the treatment of students based on race in the Charlotte Mecklenburg school system. The [Civil Rights Data Collection](https://ocrdata.ed.gov/) has agregated the data used within.

This document will use a subset of the data that CRDC has collected and explain how the data is being manipulated. 

## To begin, we need to import some libraries.

This project is written in python and will use the following libraries to make the annalysis process easier:
* pandas: used for data manipulation
* numpy: handles lots of heavier math opperations
* matplotlib: responsible for displaying visualizations

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Reading the data

When the data is grabbed from CRDC's website, it is stored in a `.csv` file. CSV stands for Comma Sepperated Values which is very descriptive of how the information is stored. A CSV is essentially a text file with rows of data seperated by commas to form the columns. The program cannot dirrectly manipulate the data in this state, so it must be read into memory. 

In [4]:
full_data = pd.read_csv('Data Files\Data Files and Layouts\CRDC 2015-16 School Data.csv', encoding='windows-1252')
full_data.shape

(96360, 1836)

### What we actually have

This file is massive. The full dataset has 96,360 rows and 1,836 columns. This is much more than is required for this project since this project is only interested in schools within CMS and only interested in information about discipline disporpotionality. The full data set contains much more data than is wanted here, so much of the data needs to be removed.

--This next section of code will select only schools that are in CMS

In [9]:
cms_schools = full_data[63467:63635]
print(cms_schools.head())
cms_schools.shape

LEA_STATE  LEA_STATE_NAME    LEAID                       LEA_NAME  \
63467        NC  NORTH CAROLINA  3702970  Charlotte-Mecklenburg Schools   
63468        NC  NORTH CAROLINA  3702970  Charlotte-Mecklenburg Schools   
63469        NC  NORTH CAROLINA  3702970  Charlotte-Mecklenburg Schools   
63470        NC  NORTH CAROLINA  3702970  Charlotte-Mecklenburg Schools   
63471        NC  NORTH CAROLINA  3702970  Charlotte-Mecklenburg Schools   

       SCHID                       SCH_NAME     COMBOKEY  JJ SCH_GRADE_PS  \
63467    614          Lake Wylie Elementary  3.70297e+11  No          Yes   
63468    632         South Charlotte Middle  3.70297e+11  No           No   
63469    652  University Meadows Elementary  3.70297e+11  No          Yes   
63470    682          Irwin Academic Center  3.70297e+11  No           No   
63471    813         Crown Point Elementary  3.70297e+11  No           No   

      SCH_GRADE_KG  ... SCH_FTESERVICES_PSY SCH_FTESERVICES_SOC SCH_JJTYPE  \
63467         

(168, 1836)

Now, a list of all of the schools in the CMS district are contained in the varaible `cms_schools`. This subset of the origional data contains 168 schools, all of which are in CMS. From here analysis of this data can be started.

### What can be done with visualization?

This data set has loads of data, so the sky is the limit. Some posibilities to explore is the demographics of the school district, the divide distribution of enrollement between various grades, and so much more. 

This first cell will show the break down of total enrollement based on grade.